src/frontend/js/hooks/useRouteInfo/index.ts

```diff
@@ -0,0 +1,22 @@
+import { useLocation, useMatches } from 'react-router-dom';
+
+/**
+ * Retrieve route information
+ * Try to find the current route from all routes that match the current location.
+ *
+ * ⚠️ This hook must be used within a Router.
+ */
+const useRouteInfo = () => {
+  const location = useLocation();
+  const matches = useMatches();
+
+  const match = matches.find((m) => location.pathname === m.pathname);
+
+  if (!match) {
+    throw new Error(`No route match the location ${location.pathname}`);
```

Comment:
```suggestion
    throw new Error(`No route match the location ${location.pathname}`);
```
I didn't find the corresponding test

Reply:
As discussed, in the context where this hook can be called, this error never occurred so instead I remove this throw statement.